internal/resources/fabric/marketplace/datasources_schema.go

```diff
@@ -0,0 +1,139 @@
+package marketplace
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func fabricMarketplaceSubscriptionDataSourceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Subscription URI information",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Equinix-assigned marketplace identifier",
+		},
+		"status": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Subscription Status like; ACTIVE, EXPIRED, CANCELLED, GRACE_PERIOD",
+		},
+		"marketplace": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Marketplace like; AWS, GCP, AZURE, REDHAT",
+		},
+		"offer_type": {
+			Type:        schema.TypeString,
+			Optional:    true,
```

Comment:
PR looks good. The only issue I have is with the schema. There are fields marked optional and nested fields marked as required. None of those are inputs to the data retrieval though. Shouldn't `uuid` be the only input marked with anything else but `Computed`?

When you do the resource schema you can have them be input types but for data source this doesn't line up with the correct attribute behavior for what is used.